sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -508,8 +511,27 @@ object UnsupportedOperationChecker extends Logging {
         case Sample(_, _, _, _, child) if child.isStreaming =>
           throwError("Sampling is not supported on streaming DataFrames/Datasets")
 
-        case Window(_, _, _, child) if child.isStreaming =>
-          throwError("Non-time-based windows are not supported on streaming DataFrames/Datasets")
+        case Window(windowExpression, _, _, child) if child.isStreaming =>
+          val windowFunc = windowExpression.flatMap { e =>
+            e.collect {
+              case we: WindowExpression => toSQLStmt(we.windowFunction.toString)
+              }
+          }.mkString(", ")
+          val columnName = windowExpression.flatMap { e =>
+            e.collect {
+              case we: WindowExpression => toSQLId(e.toAttribute.sql)
+            }
+          }.mkString(", ")
+          val windowSpec = windowExpression.flatMap { e =>
+            e.collect {
+              case we: WindowExpression => toSQLStmt(we.windowSpec.sql)
+            }
+          }.mkString(", ")
```

Comment:
Let's merge all of them and return (we.windowFunction, e.toAttribute, we.windowSpec)